apps/studio/src/server/modules/page/page.service.ts

```diff
@@ -42,3 +46,49 @@ export const createDefaultPage = ({
     }
   }
 }
+
+export const addNewVersion = async (siteId: number, pageId: number) => {
+  return await db.transaction().execute(async (tx) => {
+    const page = await getPageById({ siteId, resourceId: pageId })
+
+    if (!page.draftBlobId) {
+      return { error: "No drafts to publish for this page" }
+    }
+
+    let newVersionNum = 1
+    if (page.versionId) {
+      const currentVersion = await getVersionById({
+        versionId: page.versionId,
+      })
+      newVersionNum = Number(currentVersion.versionNum) + 1
```

Comment:
issue: continuing with the previous comment, this means that the versions service should perform inserts depending on what it finds to be the current latest version number for the given resource. That way, if we have a race condition, we don't accidentally have two inserts with the same version number into the database.

Effectively what happens is that the versions service is the one that handles the versioning procedure, and the page service is the one that is just concerned with getting the latest version from the versions service.